subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/JavaLibraryPlugin.java

```diff
@@ -46,16 +48,38 @@ public void execute(CopySpec copy) {
             }
         });
 
+        final File dependenciesFile = new File(project.getBuildDir(), "dependency-info.json");
+
+        Task createDependencyInfoFileTask = TaskMaker.task(project, "createDependencyInfoFile", CreateDependencyInfoFileTask.class, new Action<CreateDependencyInfoFileTask>() {
+            @Override
+            public void execute(final CreateDependencyInfoFileTask task) {
+                task.setDescription("Creates file with resolved runtime dependencies.");
+                task.setOutputFile(dependenciesFile);
+                task.setConfiguration(project.getConfigurations().getByName("runtime"));
+                task.setCurrentProjectVersion(project.getVersion().toString());
+
+                DeferredConfiguration.deferredConfiguration(project, new Runnable() {
+                    @Override
+                    public void run() {
+                        task.setProjectGroup(project.getGroup().toString());
+                    }
+                });
+            }
+        });
+
         ((Jar) project.getTasks().getByName("jar")).with(license);
 
         final Jar sourcesJar = project.getTasks().create(SOURCES_JAR_TASK, Jar.class, new Action<Jar>() {
             public void execute(Jar jar) {
                 jar.from(JavaPluginUtil.getMainSourceSet(project).getAllSource());
                 jar.setClassifier("sources");
                 jar.with(license);
+                jar.getMetaInf().from(dependenciesFile);
```

Comment:
Nitpick: You can also inline dependenciesFile variable and use createDependencyInfoFileTask.getOutputFile() here. This helps getting the code simpler - less variables to remember about.